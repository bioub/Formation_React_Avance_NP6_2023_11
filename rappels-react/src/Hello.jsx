function Hello() {
  return <div className="Hello">Hello, world!</div>;
}

export default Hello;
