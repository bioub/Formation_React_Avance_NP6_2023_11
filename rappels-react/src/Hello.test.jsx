import { expect, test } from 'vitest';
// import { createRoot } from 'react-dom/client';
import { renderIntoDocument } from 'react-dom/test-utils';
// import { act, create } from 'react-test-renderer';
import { render, screen } from '@testing-library/react'
import Hello from './Hello';


// test('Hello component', () => {
//   const rootEl = document.createElement('div');
//   createRoot(rootEl).render(<Hello />);
// });

// test('Hello component', () => {
//   renderIntoDocument(<Hello />);
// });

// test('Hello component', () => {
//   const rootEl = document.createElement('div');
//   act(() => {
//     createRoot(rootEl).render(<Hello />);
//   });
//   expect(rootEl.innerText).toContain('Hello, world!');
// });

// test('Hello component snapshot testing', () => {
//   expect(create(<Hello />).toJSON()).toMatchSnapshot();
// });

test('Hello component renders', () => {
  render(<Hello />);
});

test('Hello component renders Hello, world!', () => {
  render(<Hello />);
  expect(screen.queryByText('Hello, world!')).toBeInTheDocument();
});
