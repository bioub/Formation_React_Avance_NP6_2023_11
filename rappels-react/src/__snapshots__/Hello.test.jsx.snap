// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Hello component snapshot testing 1`] = `
<div
  className="Hello"
>
  Hello, world!
</div>
`;
